source/source_integration_test.go

```diff
@@ -0,0 +1,536 @@
+)
+
+const (
+	table            = "CONDUIT_SOURCE_INTEGRATION_TABLE"
+	queryCreateTable = `
+	CREATE TABLE CONDUIT_SOURCE_INTEGRATION_TABLE (
+		id int NOT NULL PRIMARY KEY,
+		cl1 VARCHAR(15),
+		cl2 CHAR,
+		cl3 CLOB,
+		cl4 LONG VARCHAR,
+		cl5 GRAPHIC(7),
+		cl6 LONG VARGRAPHIC,
+		cl7 VARGRAPHIC(15),
+		cl8 BIGINT,
+		cl9 SMALLINT,
+		cl10 DECIMAL,
+		cl11 FLOAT
+)
+	`
+	queryInsertTestData = `
+		INSERT INTO CONDUIT_SOURCE_INTEGRATION_TABLE VALUES 
+		( 1, 'varchar', 'c', 'clob', 'long varchar', 'graphic', 'long vargraphic',
+		 'vargraphic', 5455, 2321, 123.12, 123.1223),
+		( 2, 'varchar', 'c', 'clob', 'long varchar', 'graphic', 'long vargraphic', 
+		 'vargraphic', 5455, 2321, 123.12, 123.1223),
+		( 3, 'varchar', 'c', 'clob', 'long varchar', 'graphic', 'long vargraphic',
+		 'vargraphic', 5455, 2321, 123.12, 123.1223),
+		( 4, 'varchar', 'c', 'clob', 'long varchar', 'graphic', 'long vargraphic', 
+		 'vargraphic', 5455, 2321, 123.12, 123.1223)
+`
+	queryInsertCDCData = `
+		INSERT INTO CONDUIT_SOURCE_INTEGRATION_TABLE VALUES 
+		( 5, 'varchar', 'c', 'clob', 'long varchar', 'graphic', 'long vargraphic',
+		 'vargraphic', 5455, 2321, 123.12, 123.1223)
+	`
+
+	queryUpdateCDCData = `
+		UPDATE CONDUIT_SOURCE_INTEGRATION_TABLE SET CL1 ='update' 
+		WHERE ID = 5
+	`
+
+	queryDeleteCDCData = `
+		DELETE FROM CONDUIT_SOURCE_INTEGRATION_TABLE
+	`
+
+	queryDropTable         = `DROP TABLE CONDUIT_SOURCE_INTEGRATION_TABLE`
+	queryDropTrackingTable = `DROP TABLE CONDUIT_TRACKING_CONDUIT_SOURCE_INTEGRATION_TABLE`
+)
+
+func TestSource_Snapshot_Success(t *testing.T) {
+	ctx := context.Background()
+
+	cfg, err := prepareConfig()
+	if err != nil {
+		t.Skip()
+	}
+
+	err = prepareData(ctx, cfg[config.KeyConnection])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer clearData(ctx, cfg[config.KeyConnection]) // nolint:errcheck,nolintlint
+
+	s := new(Source)
+
+	err = s.Configure(ctx, cfg)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Start first time with nil position.
+	err = s.Open(ctx, nil)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// Check first read.
+	r, err := s.Read(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	// check right converting.
+	exceptedRecordPayload := sdk.Change{
+		After: sdk.RawData(`{"CL1":"varchar","CL10":"123","CL11":123.1223,"CL2":"c","CL3":"clob","CL4":"long varchar","CL5":"graphic","CL6":"long vargraphic","CL7":"vargraphic","CL8":5455,"CL9":2321,"ID":1}`), //nolint:lll// for comparing
+	}
+
+	if !reflect.DeepEqual(r.Payload, exceptedRecordPayload) {
+		t.Fatal(errors.New("wrong record payload"))
+	}
+
+	err = s.Teardown(ctx)
+	if err != nil {
+		t.Fatal(err)
+	}
+}
+
+func TestSource_Snapshot_Continue(t *testing.T) {
+	ctx := context.Background()
+
+	cfg, err := prepareConfig()
+	if err != nil {
+		t.Skip()
+	}
+
+	err = prepareData(ctx, cfg[config.KeyConnection])
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	defer clearData(ctx, cfg[config.KeyConnection]) // nolint:errcheck,nolintlint
+
+	s := new(Source)
```

Comment:
Nitpick: IMHO it makes sense to construct objects being tested in the same way they are normally used in code. That way, we are sure that the objects being tested are used in the way they are supposed to be used which makes the tests more realistic. Additionally, some setup may be run in `source.New()` which might be important.

Reply:
I am agree, I have changed